src/main/thrift/parquet.thrift

```diff
@@ -511,6 +511,57 @@ struct ColumnMetaData {
    * This information can be used to determine if all data pages are
    * dictionary encoded for example **/
   13: optional list<PageEncodingStats> encoding_stats;
+
+  /** Byte offset from beginning of file to bloom filter data. The bloom filters
+   * data of columns together is stored before the start of row group wihch describe.**/
+  14: optional i64 bloom_filter_offset;
+}
+
+/**
+  * Definition of bloom filter algorithm.
+  */
+union BloomFilterAlgorithm {
+  /** The default value 0 represents Block based bloom filter. 
+   * The bloom filter bitset is separated into tiny bucket as tiny bloom 
+   * filter, the high 32 bits hash value is used to select bucket, and 
+   * lower 32 bits hash values are used to set bits in tiny bloom filter.
+   * See “Cache-, Hash- and Space-Efficient Bloom Filters”. Specifically, 
+   * one tiny bloom filter contains eight 32-bit words (4 bytes stored in 
+   * little endian), and the algorithm set one bit in each 32-bit word.
+   *
+   * In order to set bits in bucket, the algorithm need 8 SALT values 
+   * (0x47b6137bU, 0x44974d91U, 0x8824ad5bU, 0xa2b7289dU, 0x705495c7U, 
+   * 0x2df1424bU, 0x9efc4947U, 0x5c6bfb31U) to calculate index with formular:
+   *                  index[i] = (hash * SALT[i]) >> 27 
+   **/
+   1: i32 algorithm = 0;
+}
+
+/** 
+ * Definition for hash function used to compute hash of column value.
+ * Note that the hash function take plain encoding of column values as input.
+ */
+union BloomFilterHash {
```

Comment:
This isn't quite what I meant. Have a look at the way it is done in the logical types PR. https://github.com/apache/parquet-format/pull/51/files#diff-0f9d1b5347959e15259da7ba8f4b6252R264

That uses an empty struct in the union.

Reply:
Do you mean in this way? It looks a bit strange and hard to understand from the code..